agent/consul/state/config_entry.go

```diff
@@ -832,3 +833,46 @@ func (s *Store) configEntryWithOverridesTxn(
 
 	return s.configEntryTxn(tx, ws, kind, name, entMeta)
 }
+
+// protocolForService returns the service graph protocol associated to the
+// provided service, checking all relevant config entries.
+func (s *Store) protocolForService(
+	tx *memdb.Txn,
+	ws memdb.WatchSet,
+	svc structs.ServiceID,
+) (uint64, string, error) {
+	// Get the global proxy defaults (for default protocol)
+	maxIdx, proxyConfig, err := s.configEntryTxn(tx, ws, structs.ProxyDefaults, structs.ProxyConfigGlobal, &svc.EnterpriseMeta)
+	if err != nil {
+		return 0, "", err
+	}
+
+	// Check the wildcard entry's protocol against the discovery chain protocol for the service.
```

Comment:
This comment seems outdated, since the wildcard entry's protocol isn't checked in this function.

Reply:
Yep, done.